### ISSUE TITLE
Clean up environment variable usage allowing command line options to provide values if needed

### ISSUE THREAD

**Issue by sjmudd** [vitessio/vitess#4154]
#### Feature Description

Currently to start vitess processes you need to set various environment variables to make Vitess work as needed in addition to the various parameters that each process needs. This makes some behaviour implicit on these variables being set rather than being explicit from the command line.

I think this affects the following variables:
* `VTROOT`
* `VTDATAROOT`
* `VTTOP` but this may be used internally in some scripts so may not matter.
* `VT_MYSQL_ROOT`
* `MYSQL_FLAVOR`

I know I set all of these to get vitess to behave as I need it.

If no flag exists it would be good to provide a command line option for each one and adjust the code so that the flag being set would take precedence over the environment variable setting.

Then we'd be able to start vitess processes explicitly from a clean environment without having to worry too much about setting up the environment.

Other things that might be needed:
* adjust documentation to recommend the use of the command line options rather than the environment variables
* adjust other language code (python?) to conform with the changes in Go.

I think this will make things more explicit and cleaner and perhaps later the use of the environment variables could be removed. I don't see that as being a priority right now.

**Comment by dweitzman** [comment]
On the subject of environment variables, it would probably also be worthwhile to get rid of VTROOT and avoid the assumption that git repo is checked out into src/vitess.io/vitess/, particularly since go1.11 and go modules will be available any day now.

Ideally it should be possible for test dependencies and built binaries to go in subdirectories of VTTOP.

**Comment by sjmudd** [comment]
Yes, my "testing in production environment" has no src at all and I had to adjust some stuff for the web directories for vtctld to work properly. So I think this makes sense. Build inside `VTTOP` and maybe have a `make install` target to push binaries to `$VTROOT/bin` or similar.
